### PR TITLE
Update error reporting guide

### PR DESCRIPTION
**Pull request by bruno-garcia**
The current guide on error reporting is using the last stable version of the Flutter plugin by Sentry. The latest one has huge improvements such as: 

* Sentry's Unified API.
* Complete Sentry's Protocol available.
* Automatic HTTP Breadcrumbs for http.Client
* No boilerplate for runZonedGuarded and Isolate.current.addErrorListener
* All events are enriched with Scope's Contexts, this includes Breadcrumbs, tags, User, etc...
* Automatic NavigatorObserver Breadcrumbs
* Automatic Device's Breadcrumbs through the Android and iOS SDKs or via Sentry's WidgetsBindingObserver wrapper
* No boilerplate for FlutterError.onError
* All events are enriched with Contexts's data, this includes Device's, OS, App info, etc...
* Offline caching
* Release health
* Captures not only Dart and Flutter errors, but also errors caused on the native platforms, Like Kotlin, Java, C and C++ for Android and Swift, ObjC, C, C++ for iOS
* Supports Fatal crashes, Event is going to be sent on App's restart
* Supports split-debug-info for Android only
* Flutter Android, iOS and limited support for Flutter Web

### CODE REVIEW

**Comment by bruno-garcia** [comment]
Could you please review @marandaneto?

File: src/docs/cookbook/maintenance/error-reporting.md
Comment (marandaneto) [inline src/docs/cookbook/maintenance/error-reporting.md:89]:
```suggestion
Including Swift, Objective-C, C, C++ on iOS. And on Android: Java, Kotlin, C and C++.
```

File: src/docs/cookbook/maintenance/error-reporting.md
Comment (marandaneto) [inline src/docs/cookbook/maintenance/error-reporting.md:119]: I think this could be removed as the SDK hooks up automatically

File: src/docs/cookbook/maintenance/error-reporting.md
Comment (marandaneto) [inline src/docs/cookbook/maintenance/error-reporting.md:120]:
```suggestion
[pub.dev]: {{site.pub-pkg}}/documentation/sentry_flutter/latest/sentry_flutter/sentry_flutter-library.html
```

not sure about this though

File: src/docs/cookbook/maintenance/error-reporting.md
Comment (bruno-garcia) [reply src/docs/cookbook/maintenance/error-reporting.md:89]: The thing is that it would support Rust, and other native languages, I'm not sure how much is "good enough" to list here.

File: src/docs/cookbook/maintenance/error-reporting.md
Comment (bruno-garcia) [reply src/docs/cookbook/maintenance/error-reporting.md:119]: Good point these are links and many of them are no longer needed

**Comment by bruno-garcia** [comment]
@googlebot I fixed it.

File: src/docs/cookbook/maintenance/error-reporting.md
Comment (bruno-garcia) [inline src/docs/cookbook/maintenance/error-reporting.md:48]: There's a new project type Flutter:

image

File: src/docs/cookbook/maintenance/error-reporting.md
Comment (bruno-garcia) [inline src/docs/cookbook/maintenance/error-reporting.md:49]: The installation step has a snippet which includes the DSN for the actual project:

image

File: src/docs/cookbook/maintenance/error-reporting.md
Comment (bruno-garcia) [inline src/docs/cookbook/maintenance/error-reporting.md:53]: [The `sentry` package is for pure Dart apps and the new `sentry_flutter`](https://pub.dev/packages?q=sentry) takes dependency on Flutter and integrates with it, adding native support:

image

**Comment by marandaneto** [comment]
@googlebot I fixed it.

**Review by marandaneto — APPROVED** [review]

**Comment by bruno-garcia** [comment]
Thanks @marandaneto 

Lets see if someone else from the Flutter org is able to review before I hit merge on this 🙏

File: src/docs/cookbook/maintenance/error-reporting.md
Comment (sfshaza2) [inline src/docs/cookbook/maintenance/error-reporting.md:49]:
```suggestion
  4. Copy the code snippet that includes the DSN.
```

File: src/docs/cookbook/maintenance/error-reporting.md
Comment (sfshaza2) [inline src/docs/cookbook/maintenance/error-reporting.md:79]:
```suggestion
Alternatively, you can pass the DSN to Flutter using Dart `define`:
```

File: src/docs/cookbook/maintenance/error-reporting.md
Comment (sfshaza2) [inline src/docs/cookbook/maintenance/error-reporting.md:89]:
```suggestion
This includes Swift, Objective-C, C, and C++ on iOS, and on Android: Java, Kotlin, C, and C++.
```

File: src/docs/cookbook/maintenance/error-reporting.md
Comment (sfshaza2) [inline src/docs/cookbook/maintenance/error-reporting.md:94]:
```suggestion
Besides the automatic error reporting that Sentry generates by initializing the SDK,
you can use the API to report errors to Sentry.
```

File: src/docs/cookbook/maintenance/error-reporting.md
Comment (sfshaza2) [inline src/docs/cookbook/maintenance/error-reporting.md:101]:
```suggestion
The complete API is available on [pub.dev][].
```

File: src/docs/cookbook/maintenance/error-reporting.md
Comment (sfshaza2) [inline src/docs/cookbook/maintenance/error-reporting.md:37]: Also, please remove the period from item 1.

```suggestion
  2. Import the Flutter Sentry package
  3. Initialize the Sentry SDK
  4. Capture errors programmatically
```

File: src/docs/cookbook/maintenance/error-reporting.md
Comment (sfshaza2) [inline src/docs/cookbook/maintenance/error-reporting.md:49]:
```suggestion
  4. Copy the code snippet that includes the DSN.
```

File: src/docs/cookbook/maintenance/error-reporting.md
Comment (sfshaza2) [inline src/docs/cookbook/maintenance/error-reporting.md:62]:
```suggestion
## 3. Initialize the Sentry SDK
```

File: src/docs/cookbook/maintenance/error-reporting.md
Comment (sfshaza2) [inline src/docs/cookbook/maintenance/error-reporting.md:64]:
```suggestion
Initialize the SDK to capture different unhandled errors automatically:
```

File: src/docs/cookbook/maintenance/error-reporting.md
Comment (sfshaza2) [inline src/docs/cookbook/maintenance/error-reporting.md:105]:
```suggestion
Extensive documentation about the Sentry SDK can be found on
the [Sentry's site][].
```

File: src/docs/cookbook/maintenance/error-reporting.md
Comment (sfshaza2) [inline src/docs/cookbook/maintenance/error-reporting.md:119]:
```suggestion
[Sentry's site]: https://docs.sentry.io/platforms/flutter/
```

**Review by sfshaza2 — CHANGES_REQUESTED** [review]
This LGTM once the suggestions are incorporated.

**Comment by bruno-garcia** [comment]
Thank you so much @sfshaza2! Much better.

**Review by sfshaza2 — APPROVED** [review]
LGTM!

**Pull request merged**
